content/advent-2017/tracking-the-stars.md

```diff
@@ -0,0 +1,286 @@
+) as maxts
+on gs.repo = maxts.repo and gs.ts = maxts.ts
+order by gs.stars desc;
+```
+
+![total_stars](/postimages/advent-2017/tracking-the-stars/total_stars.svg)
+
+### Total star gain
+
+All of the repositories in the tracked set gained a total of **533,614** stars over the tracking
+period. This was done by finding the min and max timestamp for each repository, getting the star
+counts at those times, finding the difference, and then summing all those differences.
+
+```sql
+select sum(maxs.stars-mins.stars) as total_increase
+from (
+    select gs.repo, gs.stars
+    from github_stars as gs
+    inner join (
+        select repo, max(ts) as ts
+        from github_stars
+        group by repo
+    ) as maxts
+    on gs.repo = maxts.repo and gs.ts = maxts.ts
+) as maxs
+inner join (
+    select gs.repo, gs.stars
+    from github_stars as gs
+    inner join (
+        select repo, min(ts) as ts
+        from github_stars
+        group by repo
+    ) as mints
+    on gs.repo = mints.repo and gs.ts = mints.ts
+) as mins
+on maxs.repo = mins.repo;
+```
+
+### Rate of star count increase
+
+Through some SQL-fu I produced the sorted list of repositories by the number of stars they collected
+per day during the time they were tracked. Hold on to your hat for this SQL statement:
+
+```sql
+select rises.repo, cast(rise as float)/cast(run as float)*(24*60*60) as stars_per_day
+from (
+    select maxs.repo, maxs.stars-mins.stars as rise
+    from (
+        select gs.repo, gs.stars
+        from github_stars as gs
+        inner join (
+            select repo, max(ts) as ts
+            from github_stars
+            group by repo
+        ) as maxts
+        on gs.repo = maxts.repo and gs.ts = maxts.ts
+    ) as maxs
+    inner join (
+        select gs.repo, gs.stars
+        from github_stars as gs
+        inner join (
+            select repo, min(ts) as ts
+            from github_stars
+            group by repo
+        ) as mints
+        on gs.repo = mints.repo and gs.ts = mints.ts
+    ) as mins
+    on maxs.repo = mins.repo
+) as rises
+inner join
+(
+    select repo, max(ts)-min(ts) as run
+    from github_stars
+    group by repo
+) as runs
+on rises.repo = runs.repo
+where runs.run > 0
+order by stars_per_day desc;
+```
+
+This produced a distribution much like others you see with this kind of data; there's a few
+repositories in the fat section of the graph with the majority of the rate increase with a long tail
+of repos that are slowly increasing. There is one unfortunate repository that had a -24 out there at
+the end.
+
+![star_gain_per_day](/postimages/advent-2017/tracking-the-stars/star_gain_per_day.svg)
+
+## Do bigger repositories grow faster?
+
+I wanted to figure out if the larger repositories grow faster than smaller ones, so I grabbed the
+number of stars at the end of the tracking period for each repository and charted that against the
+number of stars increase per day. The SQL this time as it's just a small tweak to the
```

Comment:
> The SQL this time as it's just

Unsure if you accidentally removed a word or two from here, but I'm having a hard time figuring out this sentence.

Reply:
It was an editing snafu.